runtime/common/src/main/kotlin/kotlinx/serialization/json/internal/JsonReader.kt

```diff
@@ -37,6 +39,14 @@ internal const val STRING_ESC = '\\'
 internal const val INVALID = 0.toChar()
 internal const val UNICODE_ESC = 'u'
 
+internal const val NUMBER_NEG = '-'
+internal const val NUMBER_POS = '+'
+internal const val NUMBER_SEP = '.'
+internal const val NUMBER_EXP = 'e'
+internal val NUMBER_DIGITS = arrayOf('0', '1', '2', '3', '4', '5', '6', '7', '8', '9') // TODO: ?
```

Comment:
It is better to use `CharArray` instead of `Array<Char>` for the performance sake